src/Route.php

```diff
@@ -4,98 +4,100 @@
 
 namespace Yiisoft\Router;
 
+use Attribute;
 use InvalidArgumentException;
 use RuntimeException;
 use Stringable;
 use Yiisoft\Http\Method;
-use Yiisoft\Middleware\Dispatcher\MiddlewareDispatcher;
 
 use function in_array;
 
 /**
  * Route defines a mapping from URL to callback / name and vice versa.
  */
+#[Attribute(Attribute::TARGET_METHOD | Attribute::TARGET_CLASS | Attribute::IS_REPEATABLE)]
 final class Route implements Stringable
 {
-    private ?string $name = null;
+    private const POSSIBLE_METHODS = [
```

Comment:
```suggestion
    private const METHODS = [
```

Reply:
Suggest use `Method::ALL` instead of this constant.

Reply:
Yes. That is better.